tests/functional/PostListWithDifferentStatusCept.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+$I = new FunctionalTester( $scenario );
+$I->wantTo( 'Get posts with different post_status' );
+
+/**
+ * Clear posts table.
+ */
+$I->dontHavePostInDatabase([], true);
+
+$test_posts_statuses = ['publish', 'draft', 'future'];
+foreach ($test_posts_statuses as $post_status) {
+	// Add post for each type.
+	$I->havePostInDatabase( [
+		'post_type'    => 'post',
+		'post_status'  => $post_status,
+		'post_title'   => "test {$post_status} post",
+		'post_content' => "test {$post_status} content"
+	] );
+}
+
+/**
+ * Set the content-type so we get only posts with status published and draft.
```

Comment:
comment makes it sound like content-type is at play

Reply:
https://github.com/wp-graphql/wp-graphql/pull/439

Reply:
Thanks for fixing